### PR TITLE
TO Go tenants endpoint updates

### LINKED ISSUE #2474: The tenant api needs to implement tenancy
if the tenant tree looks like this

 -root
-- tenant a
--- tenant a.a
---- tenant a.a.a

and i have tenant a.a
then really the only tenants i’m aware of are a.a and a.a.a for a read or delete.
and if i want to create or modify a tenant, i can only give it a parent of a.a or a.a.a.

### PR DESCRIPTION
**Pull request by dangogh**
This fixes #2474

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2007/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2020/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2024/
Test PASSed.

File: traffic_ops/testing/api/conf/traffic-ops-test.conf [hunk 1]
```diff
@@ -24,12 +24,12 @@
         }
     },
     "trafficOpsDB": {
-        "dbname": "to_test",
+        "dbname": "traffic_ops",
```
[the remark this answers is not in the document]
Comment (dangogh) [reply traffic_ops/testing/api/conf/traffic-ops-test.conf]: that was unintentional -- I will remove that change..

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2027/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2034/
Test PASSed.

**Pull request merged**